code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -2368,6 +2368,122 @@
 	
 	if(prob(5))
 		M.emote(pick("twitch","blink_r","shiver"))
+		
+/datum/reagent/hypozine //syndie hyperzine
+	name = "Hypozine"
+	id = HYPOZINE
+	description = "Hypozine is an extremely effective, short lasting, muscle stimulant. Metabolises rather slowly."
+	reagent_state = LIQUID
+	color = "#C8A5DC" //rgb: 200, 165, 220
+	var/has_been_hypozined = 0
+	var/has_had_heart_explode = 0 //We've applied permanent damage.
+	var/hypozined_at = 0 //world.time
+	custom_metabolism = 0.02
+	var/oldspeed = 0
+	data = 1
+
+/datum/reagent/hypozine/reagent_deleted()
+
+	if(..())
+		return 1
+
+	if(!holder)
+		return
+	var/mob/M =  holder.my_atom
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(!has_been_hypozined || has_had_heart_explode)
+			return
+		var/timedmg = ((120 SECONDS) - data) / 10
+		dehypozine(H, timedmg * 3, 1, 0)
+
+/datum/reagent/hypozine/on_mob_life(var/mob/living/M)
+
+	if(..())
+		return 1
+	
+	M.reagents.add_reagent ("hyperzine", 0.1) //To pretend it's all okay.
+	if(ishuman(M))
+		if(data<121 && !has_been_hypozined)
+			has_been_hypozined = 1
+			has_had_heart_explode = 0 //Fuck them UP after they're done going fast.
+			oldspeed = M.movement_speed_modifier
+			
+	switch(data)
+		if(60 to 100)	//Speed up after a minute
+			if(data==60)
+				to_chat(M, "<span class='notice'>You feel faster.")
+			M.movement_speed_modifier = 1.5
+			if(prob(5))
+				to_chat(M, "<span class='notice'>[pick("Your leg muscles pulsate", "You feel invigorated", "You feel like running")].")
+		if(101 to 115)	//painfully fast
+			if(data==101)
+				to_chat(M, "<span class='notice'>Your muscles start to feel pretty hot.")
+			M.movement_speed_modifier = 2
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if(prob(10))
+					if (M.get_heart())
+						to_chat(M, "<span class='notice'>[pick("Your legs are heating up", "You feel your heart racing", "You feel like running as far as you can")]!")
+					else 
+						to_chat(M, "<span class='notice'>[pick("Your legs are heating up", "Your body is aching to move", "You feel like running as far as you can")]!")
+				H.adjustFireLoss(0.1)
+		if(116 to 120)	//traverse at a velocity exceeding the norm
+			M.movement_speed_modifier = 4
+			if(data==116)
+				to_chat(M, "<span class='danger'>Your muscles are burning up!")
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if(prob(25))
+					if (M.get_heart())
+						to_chat(M, "<span class='danger'>[pick("Your legs are burning", "All you feel is your heart racing", "You feel like you have to run away from it all")]!")
+					else 
+						to_chat(M, "<span class='danger'>[pick("Your legs are burning", "You feel like you're on fire", "You feel like you have to run away from it all")]!")
+				H.adjustToxLoss(1)
+				H.adjustFireLoss(2)
+		if(121 to INFINITY)	//went2fast
+			M.movement_speed_modifier = oldspeed
```

Comment:
Reagents are not processed while dead. `movement_speed_modifier` is a human-level var.
If someone were to get to take hypozine, die from other causes before the hypozine killed them, have their body cleared of reagents, and were then defibbed, they would never `dehypozine()`, allowing them to keep the speed forever.